.github/workflows/go-build-release.yaml

```diff
@@ -0,0 +1,64 @@
+name: Go Release Artifact Publisher
+
+on:
+  workflow_call:
+    inputs:
+      release_version:
```

Comment:
We can use a `github.ref_name` instead. 
This returns a tag alias for `release` trigger. It have to be `x.y.z` pattern.